R/backend-snowflake.R

```diff
@@ -37,3 +217,40 @@ simulate_snowflake <- function() simulate_dbi("Snowflake")
 # Link to full list: https://docs.snowflake.com/en/sql-reference/sql-all.html
 #' @export
 sql_table_analyze.Snowflake <- function(con, table, ...) {}
+
+snowflake_grepl <- function(pattern, x, ignore.case = FALSE, perl = FALSE, fixed = FALSE, useBytes = FALSE) {
+  # https://docs.snowflake.com/en/sql-reference/functions/regexp.html
+  if (any(c(perl, fixed, useBytes))) {
+    abort("`perl`, `fixed` and `useBytes` parameters are unsupported")
+  }
+  # REGEXP on Snowflaake "implicitly anchors a pattern at both ends", which
+  # grepl does not.  Left- and right-pad `pattern` with .* to get grepl-like
+  # behavior
+  if (ignore.case) {
+    sql_expr(LOWER((!!x)) %REGEXP% (".*" || LOWER((!!pattern)) || ".*"))
+  } else {
+    sql_expr(((!!x)) %REGEXP% (".*" || (!!pattern) || ".*"))
+  }
+}
+snowflake_round <- function(x, digits = 0L) {
+  digits <- as.integer(digits)
+  sql_expr(round(((!!x)) %::% FLOAT, !!digits))
+}
+
+# On Snowflake, CONCAT_WS is null if any of its arguments are null.  Paste
```

Comment:
That's the same behaviour as `str_c()` so you might want to use a different translation there.

Reply:
Good point, I'll implement this.